packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -178,115 +86,35 @@ class InputDateTimePicker extends React.Component {
-			// add new main input errors
-			.concat(nextState.errors.filter(error => INPUT_ERRORS.includes(error.code)));
-
-		if (isTimeUpdate) {
-			// to avoid having errors on untouched time elements, we check only the updated part
-			let newError;
-			switch (field) {
-				case FIELD_HOURS:
-					newError = checkHours(time.hours);
-					break;
-				case FIELD_MINUTES:
-					newError = checkMinutes(time.minutes);
-					break;
-				case FIELD_SECONDS:
-					newError = checkSeconds(time.seconds);
-					break;
-				default:
-					break;
-			}
-
-			// remove old error on updated time part
-			nextErrors = nextErrors.filter(
-				error =>
-					(field === FIELD_HOURS && !HOUR_ERRORS.includes(error.code)) ||
-					(field === FIELD_MINUTES && !MINUTES_ERRORS.includes(error.code)) ||
-					(field === FIELD_SECONDS && !SECONDS_ERRORS.includes(error.code)),
-			);
-			// add the new error on updated time part
-			if (newError) {
-				nextErrors.push(newError);
-			}
+	onBlur(event, { onReset }) {
+		onReset();
+		this.closePicker();
+		if (this.props.onBlur) {
+			this.props.onBlur(event);
 		}
-
-		this.setState({ previousErrors: this.state.errors, ...nextState, errors: nextErrors }, () => {
-			if (!this.props.formMode) {
-				this.onChange(event, 'PICKER');
-			}
-		});
-	}
-
-	onBlur(event) {
-		/*
-		 This is wrapped in a timeout because when you switch focus via clic or gesture
-		 within the picker, you will have a blur event followed by a focus event.
-		 We don't want it to trigger the onBlur and to hide the picker if a focus happens just after.
-		 So here we schedule it, and on focus we cancel it.
-		 */
-		this.blurTimeout = setTimeout(() => {
-			this.resetDate();
-			this.closePicker();
-			if (this.props.onBlur) {
-				this.props.onBlur(event);
-			}
-		});
 	}
 
 	onFocus() {
-		clearTimeout(this.blurTimeout);
 		this.openPicker();
 	}
 
-	onSubmit(event, origin) {
-		event.preventDefault();
-
-		// validation
-		// to avoid having error message change on invalid elements,
-		// we don't replace the error on those elements
-		let errors = check(this.state.date, this.state.time, this.getDateOptions());
-		errors = this.state.errors
-			.filter(({ code }) => !errors.find(error => error.code === code))
-			.concat(errors);
-
-		this.setState({ errors, errorMessage: errors[0] ? errors[0].message : '' }, () => {
-			if (!errors.length) {
-				this.onChange(event, origin);
-				this.closePicker();
-			}
-		});
-	}
-
-	onInputFocus(event, focusedId) {
-		this.setState({ focusedInput: focusedId });
-	}
-
-	getDateOptions() {
-		return {
-			dateFormat: this.props.dateFormat,
-			useTime: this.props.useTime,
-			useSeconds: this.props.useSeconds,
-			useUTC: this.props.useUTC,
-			required: this.props.required,
-		};
+	onChange(event, payload) {
+		this.props.onChange(event, payload);
+		if (
+			this.props.formMode ||
+			(!this.props.formMode && !this.props.useTime && payload.origin !== 'INPUT')
```

Comment:
Closing it while selecting a date in non formMode is added here, but if we use the input we have to either tab, esc, click outside maybe adding enter keydown to this makes sense ?

Reply:
behaviour out of scope